bskyogcard/src/util/getModerationCauseInfo.ts

```diff
@@ -0,0 +1,82 @@
+import {ModerationCause} from '@atproto/api'
+
+import {CircleInfo} from '../components/icons/CircleInfo.js'
+import {EyeSlash} from '../components/icons/EyeSlash.js'
+import {IconProps} from '../components/icons/types.js'
+import {Warning} from '../components/icons/Warning.js'
+import {ModeratorData} from '../data/getModeratorData.js'
+
+export type ModerationCauseInfo = {
+  icon: React.ComponentType<IconProps>
+  name: string
+  description?: string
+  source: string
+  sourceType: 'label' | string
+  sourceDid: string
+}
+
+const globalLabelStrings = {
+  '!hide': {
+    name: `Content Blocked`,
+    description: `This content has been hidden by the moderators.`,
+  },
+  '!warn': {
+    name: `Content Warning`,
+    description: `This content has received a general warning from moderators.`,
+  },
+  '!no-unauthenticated': {
+    name: `Sign-in Required`,
+    description: `This user has requested that their content only be shown to signed-in users.`,
+  },
+  porn: {
+    name: `Adult Content`,
+    description: `Explicit sexual images.`,
+  },
+  sexual: {
+    name: `Sexually Suggestive`,
+    description: `Does not include nudity.`,
+  },
+  nudity: {
+    name: `Non-sexual Nudity`,
+    description: `E.g. artistic nudes.`,
+  },
+  'graphic-media': {
+    name: `Graphic Media`,
+    description: `Explicit or potentially disturbing media.`,
+  },
+}
+
+export function getModerationCauseInfo({
```

Comment:
This is a pretty simplistic take on this functionality. It really only resolves the name, description, and icon for now. But we could pretty easily expand this to include more info, such as who applied the label, the labeler avi, etc.